### DISCUSSION TITLE
How to resolve cross stack dependencies during aws cdk test

### DISCUSSION THREAD

**Discussion by protyay** [aws/aws-cdk#27431] [Q&A]
Hey folks, is there a way to mock certain dependencies of a stack in cdk tests ?
Let's say i've an exported value from DNSStack which is a certificate.
And this is required in the ApiGw stack, so the stack props has a cert parameter.
Now, when testing the ApiGw stack, how do i mock the cert parameter ? If i want to create a new mock Certificate , then I'm running into a lot of issues.
Sample code
```
const mockApp = new App();
const mockStack = new Stack(mockApp, 'CertStack');  
const mockCert = new Certificate(mockStack, 'cert', { domainName: '' });
  const stack = new ServiceStack(mockApp, 'service', {
   ...otherProps
    domainCert: mockCert,
  });
  ```
**Error**
Stack "service" cannot reference {CertStack/cert/Resource[Ref]} in stack "CertStack". Cross stack references are only supported for stacks deployed to the same account or between nested stacks and their parent stack (edited)

**Comment by software-artificer** [comment]
Try passing the env with the same `account` and `region` properties to each stack — that should work.
